src/odemis/util/transform.py

```diff
@@ -288,6 +288,65 @@ def to_physical_space(ji, shape, pixel_size=None):
     return xy
 
 
+def to_pixel_index(xy, shape, pixel_size=None):
+    """
+    Converts a coordinate in physical space into an image pixel index.
+
+    Inverse of `to_physical_space`. The function `to_pixel_index` converts a
+    coordinate in physical space into an image pixel index. The columns of the
+    image are aligned with the x-axis, and the rows with the y-axis. The
+    direction of the y-axis is opposite to the row-index. The origin of the
+    image pixel index is placed at the center of the top-left pixel.
+
+    Parameters
+    ----------
+    xy : tuple, list of tuples, ndarray
+        Physical coordinates, list of coordinates, or array of coordinates. For
+        each coordinate the first entry is the `x`-coordinate and the second
+        entry is the `y`-coordinate.
+    shape : tuple of ints
+        Shape of the image. The first entry is the number of rows, the second
+        entry is the number of columns in the image.
+    pixel_size : tuple of 2 floats, float (optional)
+        Pixel size in (x, y). For square pixels, a single float can be
+        provided.
```

Comment:
Let's mention that if not specified, it defaults to 1? Or maybe instead of using `None` as default value, you could use `1`, and have a special/fast case for that value.